## Experimental design

| sample    | group | condition | experiment |
|-----------|-------|-----------|------------|
| untreated |       | Control   | 1          |
| treated   |       | CLD       | 1          |
| Z4        | WT    | Control   | 2          |
| Z6        | WT    | CLD       | 2          |

## Methods

10X files generated by CellRanger for the TC-1 lung carcinoma
validation set were loaded into R 4.2.0 Patched (2022-05-13 r82357)
using the read10xCounts() function from the DropletUtils
(10.1038/s41467-018-05083-x) package version 1.16.0. Quality metrics
were computed using the scuttle package
(10.1093/bioinformatics/btw777) version 1.6.3 and cells with over 10%
of mitochondrial gene counts were removed. Counts where then
normalisation by deconvolution using the scran package
(10.12688/f1000research.9501.2) version 1.24.1 and the 1000 most
variable genes were selected based on their per-gene variance as
implemented in scran. Clusters were identified using scran's
clusterCells() function applying a two-step procedure, starting with a
vector quantisation with k-means on the PCA-reduced data (200 centres
and 100 iterations) to obtain representative centroids that are then
subjected to nearest-neighbour graph-based clustering as described
above, using k = 10 nearest neighbours. Cell type annotation was
performed using the SingleR package as above, using the MC38-OVA colo

## Session information

```
R version 4.2.0 Patched (2022-05-13 r82357)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Manjaro Linux

Matrix products: default
BLAS:   /usr/lib/libblas.so.3.10.1
LAPACK: /usr/lib/liblapack.so.3.10.1

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C
 [9] LC_ADDRESS=C               LC_TELEPHONE=C
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods
[8] base

other attached packages:
 [1] bluster_1.6.0               SingleR_1.10.0
 [3] celldex_1.6.0               pheatmap_1.0.12
 [5] robustbase_0.95-0           forcats_0.5.2
 [7] stringr_1.4.1               dplyr_1.0.10
 [9] purrr_0.3.4                 readr_2.1.2
[11] tidyr_1.2.1                 tibble_3.1.8
[13] tidyverse_1.3.2             AnnotationHub_3.4.0
[15] BiocFileCache_2.4.0         dbplyr_2.2.1
[17] scran_1.24.1                scater_1.24.0
[19] ggplot2_3.3.6               scuttle_1.6.3
[21] DropletUtils_1.16.0         SingleCellExperiment_1.18.0
[23] SummarizedExperiment_1.26.1 Biobase_2.56.0
[25] GenomicRanges_1.48.0        GenomeInfoDb_1.32.4
[27] IRanges_2.30.1              S4Vectors_0.34.0
[29] BiocGenerics_0.42.0         MatrixGenerics_1.8.1
[31] matrixStats_0.62.0

loaded via a namespace (and not attached):
  [1] readxl_1.4.1                  backports_1.4.1
  [3] igraph_1.3.5                  BiocParallel_1.30.3
  [5] digest_0.6.29                 htmltools_0.5.3
  [7] viridis_0.6.2                 fansi_1.0.3
  [9] magrittr_2.0.3                memoise_2.0.1
 [11] ScaledMatrix_1.4.1            googlesheets4_1.0.1
 [13] cluster_2.1.4                 tzdb_0.3.0
 [15] limma_3.52.3                  Biostrings_2.64.1
 [17] modelr_0.1.9                  R.utils_2.12.0
 [19] colorspace_2.0-3              blob_1.2.3
 [21] rvest_1.0.3                   rappdirs_0.3.3
 [23] ggrepel_0.9.1                 haven_2.5.1
 [25] crayon_1.5.1                  RCurl_1.98-1.8
 [27] jsonlite_1.8.0                glue_1.6.2
 [29] gtable_0.3.1                  gargle_1.2.1
 [31] zlibbioc_1.42.0               XVector_0.36.0
 [33] DelayedArray_0.22.0           BiocSingular_1.12.0
 [35] Rhdf5lib_1.18.2               DEoptimR_1.0-11
 [37] HDF5Array_1.24.2              scales_1.2.1
 [39] DBI_1.1.3                     edgeR_3.38.4
 [41] Rcpp_1.0.9                    viridisLite_0.4.1
 [43] xtable_1.8-4                  dqrng_0.3.0
 [45] bit_4.0.4                     rsvd_1.0.5
 [47] metapod_1.4.0                 httr_1.4.4
 [49] RColorBrewer_1.1-3            ellipsis_0.3.2
 [51] pkgconfig_2.0.3               R.methodsS3_1.8.2
 [53] locfit_1.5-9.6                utf8_1.2.2
 [55] tidyselect_1.1.2              rlang_1.0.5
 [57] later_1.3.0                   AnnotationDbi_1.58.0
 [59] munsell_0.5.0                 BiocVersion_3.15.2
 [61] cellranger_1.1.0              tools_4.2.0
 [63] cachem_1.0.6                  cli_3.4.0
 [65] ExperimentHub_2.4.0           generics_0.1.3
 [67] RSQLite_2.2.17                broom_1.0.1
 [69] fastmap_1.1.0                 yaml_2.3.5
 [71] bit64_4.0.5                   fs_1.5.2
 [73] KEGGREST_1.36.3               sparseMatrixStats_1.8.0
 [75] mime_0.12                     R.oo_1.25.0
 [77] xml2_1.3.3                    compiler_4.2.0
 [79] beeswarm_0.4.0                filelock_1.0.2
 [81] curl_4.3.2                    png_0.1-7
 [83] interactiveDisplayBase_1.34.0 reprex_2.0.2
 [85] statmod_1.4.37                stringi_1.7.8
 [87] lattice_0.20-45               Matrix_1.5-1
 [89] vctrs_0.4.1                   pillar_1.8.1
 [91] lifecycle_1.0.2               rhdf5filters_1.8.0
 [93] BiocManager_1.30.18           BiocNeighbors_1.14.0
 [95] bitops_1.0-7                  irlba_2.3.5
 [97] httpuv_1.6.6                  R6_2.5.1
 [99] promises_1.2.0.1              gridExtra_2.3
[101] vipor_0.4.5                   codetools_0.2-18
[103] assertthat_0.2.1              rhdf5_2.40.0
[105] withr_2.5.0                   GenomeInfoDbData_1.2.8
[107] parallel_4.2.0                hms_1.1.2
[109] grid_4.2.0                    beachmat_2.12.0
[111] DelayedMatrixStats_1.18.0     googledrive_2.0.0
[113] shiny_1.7.2                   lubridate_1.8.0
[115] ggbeeswarm_0.6.0
```
